CoreML compile error

**Describe the bug**
I'm unable to compile with --use_coreml due to warnings being treated as errors.  This happens in the current code as well as the last releases.

**Urgency**
None

**System information**
- OS Platform and Distribution: macOS 12.3
- ONNX Runtime installed from source
- ONNX Runtime version: 1.11.0
- Python version: 3.9.10
- GCC/Compiler version (if compiling from source): Apple clang version 13.1.6 (clang-1316.0.21.2)

**To Reproduce**
./build.sh --use_coreml --osx_arch x86_64 --parallel --config ReleaseWithDebInfo
Fails with:
onnxruntime/core/providers/coreml/builders/impl/pool_op_builder.cc:45:8: error: 
      variable 'is_average_pool' set but not used
      [-Werror,-Wunused-but-set-variable]
  bool is_average_pool = false;

**Expected behavior**
Removing all references to 'is_average_pool' in pool_op_builder.cc will allow compilation to complete.  Is this variable needed for future development?

--- a/onnxruntime/core/providers/coreml/builders/impl/pool_op_builder.cc
+++ b/onnxruntime/core/providers/coreml/builders/impl/pool_op_builder.cc
@@ -42,7 +42,6 @@ Status PoolOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
   const auto& input_defs = node.InputDefs();
 
   bool is_global_pooling = false;
-  bool is_average_pool = false;
   if (op_type == "GlobalAveragePool") {
     is_global_pooling = true;
     coreml_pool->set_type(COREML_SPEC::PoolingLayerParams_PoolingType_AVERAGE);
@@ -50,7 +49,6 @@ Status PoolOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
     is_global_pooling = true;
     coreml_pool->set_type(COREML_SPEC::PoolingLayerParams_PoolingType_MAX);
   } else if (op_type == "AveragePool") {
-    is_average_pool = true;
     coreml_pool->set_type(COREML_SPEC::PoolingLayerParams_PoolingType_AVERAGE);
   } else if (op_type == "MaxPool") {
     coreml_pool->set_type(COREML_SPEC::PoolingLayerParams_PoolingType_MAX);

